### ISSUE TITLE
Fetch adapter lost session in cross-domain

### ISSUE THREAD

**Issue by miyagipipi** [axios/axios#6475]
### Describe the bug

When I used Axios's fetch adapter and involved cross domain sessions, I found that the fetch adapter lost session every time it requested. 
This cannot maintain the same session. 
I see that the source code fetch.js does not have "credentials" and "mode" set, which are important configuration contents in the fetch. 
Is it possible to add these configurations in the fetch adapter?
In simpler terms, it's like this in fetch.js:
<img width="599" alt="06ed52a395b7b45aae95525daf91fab" src="https://github.com/axios/axios/assets/66454376/cb86db37-36d0-4331-abdb-64c2a460da6d">
<img width="473" alt="94ec51cbe5e979900def26f2ac27b8a" src="https://github.com/axios/axios/assets/66454376/31e60e6e-82f8-4bd1-bcb7-44567c3fb83a">

### Axios Version

1.7.2

### Adapter Version

fetch.js

**Comment by sasanazizi** [comment]
same issue.
im using fetch adapter so is used **fetchOptions** against  `withCredentials: true` config like:

`fetchOptions: { credentials: 'same-origin' }`
 
but no cookies are sent to the backend.

**Comment by miyagipipi** [comment]
My first picture was a bit incorrect. I was just trying to express my hope that these options could be changed to configurable in the fetch adapter.

**Comment by jasonsaayman** [comment]
i believe this is much improved in the latest version, please retry with the latest version and should the problem persist open a new issue and mention me
